core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -996,12 +995,6 @@ private[spark] class BlockManager(
         // notified the master about the availability of this block, so we need to send an update
         // to remove this block location.
         removeBlockInternal(blockId, tellMaster = tellMaster)
-        // The `putBody` code may have also added a new block status to TaskMetrics, so we need
```

Comment:
I'm having flashbacks to how tricky it was to figure out this logic when refactoring the block manager code (this used to be a lot messier). Really happy to see this get removed.